services/github/src/api.js

```diff
@@ -389,6 +389,60 @@ builder.declare({
   }
 });
 
+builder.declare({
+  name: 'checksBadge',
+  title: 'Latest Build Checksuite Status Badge',
+  description: [
+    'Checks the status of the latest checksuite of a given ref',
+    'and returns corresponding badge svg.',
+    'The ref can be a branch, a tag or SHA',
+  ].join('\n'),
+  stability: 'experimental',
+  method: 'get',
+  route: '/repository/:owner/:repo/:ref/checksBadge.svg',
+}, async function(req, res) {
+  // Extract owner, repo and branch from request into variables
+  let {owner, repo, ref} = req.params;
+
+  // This has nothing to do with user input, so we should be safe
+  let fileConfig = {
+    root: __dirname + '/../assets/',
+    headers: {
+      'Cache-Control': 'no-cache, no-store, must-revalidate',
+    },
+  };
+
+  let instGithub = await installationAuthenticate(owner, this.OwnersDirectory, this.github);
+
+  if (instGithub) {
+    try {
+      const checksuites = await instGithub.checks.listSuitesForRef({
+        owner,
+        repo,
+        ref,
+        app_id: this.cfg.github.credentials.integrationId,
+      });
```

Comment:
What if this is a private repo - are we leaking data about a private repo's status?  I'm not sure if this is worth worrying about or not.. Certainly, the tasks that led to the status are public..

Reply:
What do you mean by "leaking data"?

Reply:
Are we provided a public way to get data that GitHub otherwise keeps secret?

Reply:
Don't we already do that with the existing badges API?